File: src/main/java/com/ploy/bubble_server_v3/domain/reservation/domain/repository/ReservationRepository.java
package com.ploy.bubble_server_v3.domain.reservation.domain.repository;

import com.ploy.bubble_server_v3.domain.reservation.domain.Reservation;
import com.ploy.bubble_server_v3.domain.reservation.exception.ReservationNotFoundException;
import com.ploy.bubble_server_v3.domain.user.domain.Users;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDateTime;
import java.util.List;

public interface ReservationRepository extends JpaRepository<Reservation, Long> {

    List<Reservation> findByUser(Users user);

    List<Reservation> findByResDateBetween(LocalDateTime start, LocalDateTime end);

    default Reservation getById(Long reservationId) {
        return findById(reservationId)
                .orElseThrow(() -> new ReservationNotFoundException(reservationId));
    }
}

File: src/main/kotlin/com/ploy/bubble_server_v2/domain/washingMachine/repository/ReservationRepository.kt
package com.ploy.bubble_server_v2.domain.washingMachine.repository

import com.ploy.bubble_server_v2.domain.washingMachine.model.entity.Reservation

import org.springframework.data.jpa.repository.JpaRepository

interface ReservationRepository : JpaRepository<Reservation?, Long?> {
    fun findByUserIdAndDate(userId: Long?, date: java.time.LocalDate?): java.util.Optional<Reservation?>?
    fun findByWashingRoomAndDateBetween(
        washingRoom: String?,
        startDate: java.time.LocalDate?,
        endDate: java.time.LocalDate?
    ): List<Reservation?>?

    fun findByUserIdAndDateBetween(
        userId: Long?,
        startDate: java.time.LocalDate?,
        endDate: java.time.LocalDate?
    ): List<Reservation?>?

    fun findByUserId(userId: Long?): List<Reservation?>?
    fun findByUserIdAndDateAndIsCancelFalse(
        userId: Long?,
        date: java.time.LocalDate?
    ): java.util.Optional<Reservation?>?
}